app/src/main/java/com/example/levi9application/FavoritesFragment.kt

```diff
@@ -10,39 +10,46 @@ import androidx.recyclerview.widget.GridLayoutManager
 import com.example.levi9application.databinding.FragmentFavoritesBinding
 import com.example.levi9application.model.FavoriteItem
 import com.example.levi9application.viewModel.FavoritesViewModel
+import dagger.hilt.android.AndroidEntryPoint
 
-
-class FavoritesFragment : Fragment(R.layout.fragment_favorites){
+@AndroidEntryPoint
+class FavoritesFragment : Fragment(R.layout.fragment_favorites) {
     private lateinit var _binding: FragmentFavoritesBinding
     private lateinit var viewModel: FavoritesViewModel
     private lateinit var _adapter: FavoritesAdapter
-    private lateinit var list: MutableList<FavoriteItem>
+    private lateinit var list: List<FavoriteItem>
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
+    ): View {
 
         _binding = FragmentFavoritesBinding.inflate(inflater, container, false)
         viewModel = ViewModelProvider(this)[FavoritesViewModel::class.java]
-        list = mutableListOf()
-        setList()
         val layoutManager = GridLayoutManager(context, 2)
+
+        list = mutableListOf()
```

Comment:
Move everything that is related to views inside the onViewCreated callback. Here the app might crash since we are accessing binding that may not be initialized.